ethcore/light/src/on_demand/mod.rs

```diff
@@ -54,13 +54,13 @@ struct Peer {
 }
 
 impl Peer {
-	// whether this peer can fulfill the
+	// whether this peer can fulfill the necessary capabilities
 	fn can_fulfill(&self, c: &Capabilities) -> bool {
 		let caps = &self.capabilities;
 
-		caps.serve_headers == c.serve_headers &&
-			caps.serve_chain_since >= c.serve_chain_since &&
-			caps.serve_state_since >= c.serve_chain_since
+		caps.serve_headers >= c.serve_headers &&
+			(c.serve_chain_since.is_none() || caps.serve_chain_since <= c.serve_chain_since) &&
+			(c.serve_state_since.is_none() || caps.serve_state_since <= c.serve_state_since)
```

Comment:
Why does `is_none` allow a request?
Shouldn't  chain and state numbers be strictly lower?